### ISSUE TITLE
Solver does not always select the latest version

### ISSUE THREAD

**Issue by gaborcsardi** [r-lib/pkgdepends#212]
```r
options(repos = c(CRAN = "https://cloud.r-project.org",
                  ncov = "https://ncov-ic.github.io/drat/"))
proposal <- pkgdepends::new_pkg_installation_proposal("sircovid")
proposal$solve()
proposal$draw()
```

This selects sircovid 0.2.0, even though a newer one is available in another repo (?), and both need compilation.

**Comment by richfitz** [comment]
"another repo" here just meaning the cran-like repository "https://ncov-ic.github.io/drat/" (rather than a git repository)

**Comment by richfitz** [comment]
I'm using the workaround of "upgrade" as suggested, but that has the side-effect of *always* updating the sircovid package. If I do

```
options(repos = c(CRAN = "https://cloud.r-project.org",
                  ncov = "https://ncov-ic.github.io/drat/"))
proposal <- pkgdepends::new_pkg_installation_proposal(
  "sircovid", policy = "upgrade")
proposal$solve()
proposal$download()
proposal$install()

proposal <- pkgdepends::new_pkg_installation_proposal(
  "sircovid", policy = "upgrade")
proposal$solve()
proposal$draw()
```

I see

```
sircovid 0.6.1 -> 0.6.1 [upd][bld][cmp][dl] (unknown size)
├─mcstate 0.2.13 
│ ├─R6 2.4.1 
│ ├─dust 0.4.13 
│ │ ├─R6
│ │ ├─cpp11 0.2.3 
│ │ ├─glue 1.4.2 
│ │ ├─pkgbuild 1.1.0 
│ │ │ ├─callr 3.5.1 
```

as it tries to reinstall sircovid which is unchanged

**Comment by gaborcsardi** [comment]
https://github.com/r-lib/pkgdepends/commit/1e86bd5e3afc0f2512b787a7c5b4d9ba8878b618 addresses the re-installation for the `"upgrade"` issue. 

It happened, because we only trusted version numbers for CRAN and Bioconductor packages. This is because in other cases people often update packages without changing the version number. 

Now we also accept packages that were installed from another CRAN-like repository.

**Comment by gaborcsardi** [comment]
The original issue happens because the lazy policy tried to do as little as possible, and the older sircovid version has less dependencies, so it chooses that. 

This is, however, clearly not what most users want, so we need to change it. I think implementing #178 will probably solve this issue as well, because that will require installing the latest available version if the package was specified for installation. (I.e. it is not a dependency of the requested packages.) I'll fix #178 soon, and then come back to this issue.

**Comment by gaborcsardi** [comment]
OK, I think the original issue is now fixed as well by https://github.com/r-lib/pkgdepends/commit/87df1e2b93a767f3954a97ceb77f7461ac8e04f9. 

Please reopen is you still have problems.
